Segmentation violation - terraform-provider-azurerm_v4.5.0_x5 plugin crashed

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.8

### AzureRM Provider Version

4.5.0

### Affected Resource(s)/Data Source(s)

azurerm_subscription_policy_assignment

### Terraform Configuration Files

```hcl
resource "azurerm_subscription_policy_assignment" "this" {
  for_each = { for k, v in var.policies : v["name"] => v }

  name                 = each.key
  display_name         = each.value["display_name"]
  policy_definition_id = "/providers/Microsoft.Authorization/policyDefinitions/${each.value["id"]}"
  subscription_id      = "/subscriptions/${var.subscription_id}"
  parameters           = try(jsonencode(each.value["parameters"]), null)
}
# called about ~3000 times, about 50 calls per Azure subscription each.
```


### Debug Output/Panic Output

```shell
╷
│ Error: Request cancelled
│ 
│   with module.subgroup["XXXXXXXXXXXXXXXXXX"].module.subscription_policies["XXXXXXXXXXXXXX"].azurerm_subscription_policy_assignment.this["vmss-patching"],
│   on modules/subscription-policy/main.tf line 1, in resource "azurerm_subscription_policy_assignment" "this":
│    1: resource "azurerm_subscription_policy_assignment" "this" {
│ 
│ The plugin.(*GRPCProvider).ValidateResourceConfig request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.subgroup["XXXXXX"].module.subscription_policies["XXXXXXXX"].azurerm_subscription_policy_assignment.this["aks-priv-containers"],
│   on modules/subscription-policy/main.tf line 1, in resource "azurerm_subscription_policy_assignment" "this":
│    1: resource "azurerm_subscription_policy_assignment" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│   with module.subgroup["XXXXXXXXXXXX"].module.subscription_policies["XXXXXXXXXXXXXXXXXXXXX"].azurerm_subscription_policy_assignment.this["aks-naked-pods"],
│   on modules/subscription-policy/main.tf line 1, in resource "azurerm_subscription_policy_assignment" "this":
│    1: resource "azurerm_subscription_policy_assignment" "this" {
│ 
│ The plugin.(*GRPCProvider).ValidateResourceConfig request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│   with module.subgroup["XXXXXXXXXXXXXX"].module.subscription_policies["XXXXXXXXXXXXX"].azurerm_subscription_policy_assignment.this["disk-priv-link"],
│   on modules/subscription-policy/main.tf line 1, in resource "azurerm_subscription_policy_assignment" "this":
│    1: resource "azurerm_subscription_policy_assignment" "this" {
│ 
│ The plugin.(*GRPCProvider).ValidateResourceConfig request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v4.5.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5f8ac4b]

goroutine 7527 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/policy.waitForPolicyAssignmentToStabilize.func1()
	github.com/hashicorp/terraform-provider-azurerm/internal/services/policy/assignment.go:40 +0x10b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/retry/state.go:113 +0x1b0
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext in goroutine 160
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/retry/state.go:86 +0x23b

Error: The terraform-provider-azurerm_v4.5.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

The about 2000-3000 Azure Policies should have been created and saved to the Terraform state.

### Actual Behaviour

The policies were created, but not saved to the state since the plugin crashed. They had to be imported to the state manually later.

### Steps to Reproduce

- create reasonable big configuration that applies Azure Policies (policies can be provided as YAML later if needed)
- try to apply it

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/policy/assignment.go
+++ b/internal/services/policy/assignment.go
@@ -37,7 +37,13 @@ func waitForPolicyAssignmentToStabilize(ctx context.Context, client *assignments
 					return resp, strconv.Itoa(resp.HttpResponse.StatusCode), nil
 				}
 
-				return nil, strconv.Itoa(resp.HttpResponse.StatusCode), fmt.Errorf("polling for %s: %+v", id, err)
+				// The `HttpResponse` might be nil, reported on https://github.com/hashicorp/terraform-provider-azurerm/issues/27693
+				if resp.HttpResponse != nil {
+					return nil, strconv.Itoa(resp.HttpResponse.StatusCode), fmt.Errorf("polling for %s: %+v", id, err)
+				}
+
+				return nil, "0", fmt.Errorf("polling for %s: %+v", id, err)
+
 			}
 
 			return resp, strconv.Itoa(resp.HttpResponse.StatusCode), nil

